crates/ark/src/modules/positron/options.R

```diff
@@ -29,20 +29,19 @@ options(device = function() {
 # Set cran mirror
 repos <- getOption("repos")
 rstudio_cran <- "https://cran.rstudio.com/"
-
 if (is.null(repos) || !is.character(repos)) {
-    options(repos = c(CRAN = rstudio_cran))
+    repos <- c(CRAN = rstudio_cran)
 } else {
     if ("CRAN" %in% names(repos)) {
         if (identical(repos[["CRAN"]], "@CRAN@")) {
             repos[["CRAN"]] <- rstudio_cran
-            options(repos = repos)
         }
     } else {
         repos <- c(CRAN = rstudio_cran, repos)
-        options(repos = repos)
     }
 }
+attr(repos, "Positron") <- TRUE
```

Comment:
We should not unconditionally set this. IIUC, we should only set this if we detect the "default" repos behavior, i.e. if we see `"@CRAN@"` as the `repos[["CRAN"]]` value

That is what RStudio does

https://github.com/rstudio/rstudio/blob/078e21116b0e34aff92addf961699017adb62fc5/src/cpp/r/R/Tools.R#L682-L712

I _think_ that means you only set the attribute in the `if (identical(repos[["CRAN"]], "@CRAN@")) {` branch

---

In theory this allows a user to set their own `repos` in their `.Rprofile`, and in that case we are not supposed to mark that with a `"Positron"` attribute, as that did not come from us. In practice that might not work, because I _think_ this code might get run before we source a user's `.Rprofile` on their behalf, but that would be another bug to fix separately, I think.

Reply:
Ah, I misunderstood before how this was being used! Thanks 👍